File: UniversityApi.API/Contracts/ICoursesRepository.cs
using Microsoft.EntityFrameworkCore;
using UniversityApi.API.DataBase.Entities;

namespace UniversityApi.API.Contracts
{
    public interface ICoursesRepository : IGenericRepository<EntityCourse>
    {
        Task<EntityCourse> UpdateWithProfessorId(EntityCourse course, Guid professorId);
        Task<bool> CourseCodeIsOccupied(string courseCode);
        Task<EntityCourse> AddWithProfessorId(EntityCourse entity, Guid professorId);
    }
}

File: FsUniversityApi/Contracts/ICoursesRepository.fs
namespace FsUniversityApi.Contracts

open FsUniversityApi.Database.Entities.StudentCourseProfessor
open System.Threading.Tasks
open System

type ICoursesRepository =
    abstract member UpdateWithProfessorId : Course * Guid -> Task<Course>
    abstract member CourseCodeIsOccupied : string -> Task<bool>
    abstract member AddWithProfessorId : Course * Guid -> Task<Course>